src/toil/lib/threading.py

```diff
@@ -313,10 +313,10 @@ def process_name_exists(base_dir: str, name: str) -> bool:
 
     nameFD = None
     try:
-        # Otherwise see if we can lock it shared, for which we need an FD, but
-        # only for reading.
-        nameFD = os.open(nameFileName, os.O_RDONLY)
         try:
+            # Otherwise see if we can lock it shared, for which we need an FD, but
+            # only for reading.
+            nameFD = os.open(nameFileName, os.O_RDONLY)
             fcntl.lockf(nameFD, fcntl.LOCK_SH | fcntl.LOCK_NB)
         except OSError as e:
```

Comment:
```suggestion
            # Otherwise see if we can lock it shared, for which we need an FD, but
            # only for reading.
            nameFD = os.open(nameFileName, os.O_RDONLY)
            fcntl.lockf(nameFD, fcntl.LOCK_SH | fcntl.LOCK_NB)
        except FileNotFoundError as e:
            # File has vanished
            return False
        except OSError as e:
```